dlib/image_processing/full_object_detection.h

```diff
@@ -183,6 +183,60 @@ namespace dlib
             item.label = "";
     }
 
+// ----------------------------------------------------------------------------------------
+
+    struct yolo_rect
+    {
+        yolo_rect() = default;
+        yolo_rect(const drectangle& r) : rect(r) {}
+        yolo_rect(const drectangle& r, double score) : rect(r),detection_confidence(score) {}
+        yolo_rect(const drectangle& r, double score, const std::string& label) : rect(r),detection_confidence(score), label(label) {}
+        yolo_rect(const mmod_rect& r) : rect(r.rect), detection_confidence(r.detection_confidence), ignore(r.ignore), label(r.label) {}
+
+        drectangle rect;
```

Comment:
I made this a `drectangle` for no particular reason... it was originally a `rectangle`, but when the loss was not working, I thought it might be due to some missing cast or something (I was a bit desperate) so I made it a `drectangle` to make sure that was not causing any problems.

However, I don't know if it makes a lot of sense to have sub-pixel accuracy for the bounding box. If you think It should be a `rectangle`, I will change it back and add the proper casts in the loss function.

Reply:
Na `drectangle` seems like the right call.